libs/platform/src/lib/components/menu/menu-trigger.directive.ts

```diff
@@ -124,7 +124,15 @@ export class MenuTriggerDirective implements OnDestroy, AfterContentInit {
     }
 
     toggleMenu() {
-        this._isMenuOpen ? this.destroyMenu() : this.openMenu();
+        /**
+         * Need to add delay here to ensure that any "closeMenu" operation which
+         * has been invoked within an "outsideClickSubscription" gets resolved
+         * before "openMenu" is called. This can happen when there are multiple
+         * triggers for the same menu.
+         */
+        setTimeout(() => {
+            this._isMenuOpen ? this.destroyMenu() : this.openMenu();
```

Comment:
Can you use tick() instead ?

Reply:
I believe `tick()` is a method from Angular's testing module:
https://angular.io/api/core/testing/tick

I don't think there exist a `tick` function in native Javascript, unless you know of some other instance.

Reply:
you are right. I though I was commenting test.